# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/SSLTokensCache.cpp
# Commit: ee9744982673
# Full Hash: ee974498267381418f06efd6ab92c1f2c5f3735d
# Author: Dana Keeler <dkeeler@mozilla.com>
# Date: 2022-11-15 16:44:51
# Regressor Bug: 1791633
# File Overlap Count: 1
# Description:
#   Bug 1791633 - separate nsITLSSocketControl from nsITransportSecurityInfo r=necko-reviewers,kershaw,jschanck
#   
#   Depends on D160311
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D160313
# ==============================================================================

diff -r f5a4bfdaba40 -r ee9744982673 netwerk/base/SSLTokensCache.cpp
--- a/netwerk/base/SSLTokensCache.cpp	Tue Nov 15 05:34:40 2022 +0000
+++ b/netwerk/base/SSLTokensCache.cpp	Tue Nov 15 05:34:40 2022 +0000
@@ -3,13 +3,14 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "SSLTokensCache.h"
+
+#include "CertVerifier.h"
+#include "CommonSocketControl.h"
+#include "TransportSecurityInfo.h"
 #include "mozilla/ArrayAlgorithm.h"
+#include "mozilla/Logging.h"
 #include "mozilla/Preferences.h"
-#include "mozilla/Logging.h"
 #include "nsIOService.h"
-#include "nsNSSIOLayer.h"
-#include "TransportSecurityInfo.h"
-#include "CertVerifier.h"
 #include "ssl.h"
 #include "sslexp.h"
 
@@ -192,7 +193,7 @@
 // static
 nsresult SSLTokensCache::Put(const nsACString& aKey, const uint8_t* aToken,
                              uint32_t aTokenLen,
-                             nsITransportSecurityInfo* aSecInfo) {
+                             CommonSocketControl* aSocketControl) {
   PRUint32 expirationTime;
   SSLResumptionTokenInfo tokenInfo;
   if (SSL_GetResumptionTokenInfo(aToken, aTokenLen, &tokenInfo,
@@ -205,13 +206,13 @@
   expirationTime = tokenInfo.expirationTime;
   SSL_DestroyResumptionTokenInfo(&tokenInfo);
 
-  return Put(aKey, aToken, aTokenLen, aSecInfo, expirationTime);
+  return Put(aKey, aToken, aTokenLen, aSocketControl, expirationTime);
 }
 
 // static
 nsresult SSLTokensCache::Put(const nsACString& aKey, const uint8_t* aToken,
                              uint32_t aTokenLen,
-                             nsITransportSecurityInfo* aSecInfo,
+                             CommonSocketControl* aSocketControl,
                              PRUint32 aExpirationTime) {
   StaticMutexAutoLock lock(sLock);
 
@@ -223,25 +224,30 @@
     return NS_ERROR_NOT_INITIALIZED;
   }
 
-  if (!aSecInfo) {
+  if (!aSocketControl) {
     return NS_ERROR_FAILURE;
   }
+  nsCOMPtr<nsITransportSecurityInfo> securityInfo;
+  nsresult rv = aSocketControl->GetSecurityInfo(getter_AddRefs(securityInfo));
+  if (NS_FAILED(rv)) {
+    return rv;
+  }
 
   nsCOMPtr<nsIX509Cert> cert;
-  aSecInfo->GetServerCert(getter_AddRefs(cert));
+  securityInfo->GetServerCert(getter_AddRefs(cert));
   if (!cert) {
     return NS_ERROR_FAILURE;
   }
 
   nsTArray<uint8_t> certBytes;
-  nsresult rv = cert->GetRawDER(certBytes);
+  rv = cert->GetRawDER(certBytes);
   if (NS_FAILED(rv)) {
     return rv;
   }
 
   Maybe<nsTArray<nsTArray<uint8_t>>> succeededCertChainBytes;
   nsTArray<RefPtr<nsIX509Cert>> succeededCertArray;
-  rv = aSecInfo->GetSucceededCertChain(succeededCertArray);
+  rv = securityInfo->GetSucceededCertChain(succeededCertArray);
   if (NS_FAILED(rv)) {
     return rv;
   }
@@ -259,7 +265,7 @@
     }
 
     bool builtInRoot = false;
-    rv = aSecInfo->GetIsBuiltCertChainRootBuiltInRoot(&builtInRoot);
+    rv = securityInfo->GetIsBuiltCertChainRootBuiltInRoot(&builtInRoot);
     if (NS_FAILED(rv)) {
       return rv;
     }
@@ -267,27 +273,27 @@
   }
 
   bool isEV;
-  rv = aSecInfo->GetIsExtendedValidation(&isEV);
+  rv = securityInfo->GetIsExtendedValidation(&isEV);
   if (NS_FAILED(rv)) {
     return rv;
   }
 
   uint16_t certificateTransparencyStatus;
-  rv = aSecInfo->GetCertificateTransparencyStatus(
+  rv = securityInfo->GetCertificateTransparencyStatus(
       &certificateTransparencyStatus);
   if (NS_FAILED(rv)) {
     return rv;
   }
 
   nsITransportSecurityInfo::OverridableErrorCategory overridableErrorCategory;
-  rv = aSecInfo->GetOverridableErrorCategory(&overridableErrorCategory);
+  rv = securityInfo->GetOverridableErrorCategory(&overridableErrorCategory);
   if (NS_FAILED(rv)) {
     return rv;
   }
 
   Maybe<nsTArray<nsTArray<uint8_t>>> failedCertChainBytes;
   nsTArray<RefPtr<nsIX509Cert>> failedCertArray;
-  rv = aSecInfo->GetFailedCertChain(failedCertArray);
+  rv = securityInfo->GetFailedCertChain(failedCertArray);
   if (NS_FAILED(rv)) {
     return rv;
   }